jf_agent/git/github.py

```diff
@@ -289,63 +296,63 @@ def _get_merge_commit(client: GithubClient, pr, strip_text_content, redact_names
 
 
 def _normalize_pr(client: GithubClient, pr, strip_text_content, redact_names_and_urls):
-    return {
-        'id': pr['number'],
-        'author': _normalize_user(client.get_json(pr['user']['url'])),
-        'title': sanitize_text(pr['title'], strip_text_content),
-        'body': sanitize_text(pr['body'], strip_text_content),
-        'is_closed': pr['state'] == 'closed',
-        'is_merged': pr['merged'],
-        'created_at': pr['created_at'],
-        'updated_at': pr['updated_at'],
-        'closed_date': pr['closed_at'] if pr['closed_at'] else None,
-        'url': (pr['html_url'] if not redact_names_and_urls else None),
-        'base_repo': _normalize_pr_repo(pr['base']['repo'], redact_names_and_urls),
-        'base_branch': (
+    return NormalizedPullRequest(
+        id=pr['number'],
+        additions=pr['additions'],
+        deletions=pr['deletions'],
+        changed_files=pr['changed_files'],
+        is_closed=pr['state'] == 'closed',
+        is_merged=pr['merged'],
+        created_at=pr['created_at'],
+        updated_at=pr['updated_at'],
+        merge_date=pr['merged_at'] if pr['merged_at'] else None,
+        closed_date=pr['closed_at'] if pr['closed_at'] else None,
+        title=sanitize_text(pr['title'], strip_text_content),
+        body=sanitize_text(pr['body'], strip_text_content),
+        url=(pr['html_url'] if not redact_names_and_urls else None),
+        base_branch=(
             pr['base']['ref']
             if not redact_names_and_urls
             else _branch_redactor.redact_name(pr['base']['ref'])
         ),
-        'head_repo': _normalize_pr_repo(pr['head']['repo'], redact_names_and_urls),
-        'head_branch': (
+        head_branch=(
             pr['head']['ref']
             if not redact_names_and_urls
             else _branch_redactor.redact_name(pr['head']['ref'])
         ),
-        'additions': pr['additions'],
-        'deletions': pr['deletions'],
-        'changed_files': pr['changed_files'],
-        'comments': [
+        author=_normalize_user(client.get_json(pr['user']['url'])),
+        merged_by=(
+            _normalize_user(client.get_json(pr['merged_by']['url'])) if pr['merged'] else None
+        ),
+        commits=[
+            _normalize_commit(c, pr['base']['repo'], strip_text_content, redact_names_and_urls)
+            for c in tqdm(
+                client.get_pr_commits(pr['base']['repo']['full_name'], pr['number']),
+                f'downloading commits for PR {pr["number"]}',
+                leave=False,
+                unit='commits',
+            )
+        ],
+        merge_commit=_get_merge_commit(client, pr, strip_text_content, redact_names_and_urls),
+        comments=[
```

Comment:
Should we go as far as to create dataclasses to model comments / approvals? Probably not necessary, just something I was thinking about while reviewing.

Reply:
Good catch! I missed these, but we do have dataclasses to model comments and reviewws on pull requests